shared/src/ledger/ibc/vp/channel.rs

```diff
@@ -236,6 +241,13 @@ where
         tx_data: &[u8],
     ) -> Result<()> {
         let prev_channel = self.channel_end_pre(port_channel_id)?;
+        let conn_id = channel.connection_hops().get(0).ok_or_else(|| {
+            Error::InvalidChannel(format!(
+                "No connection for the channel: Port/Channel {}",
+                port_channel_id,
+            ))
+        })?;
+        let counterparty = channel.counterparty();
         match channel.state() {
```

Comment:
There is a lot of repeated code here. Can you instead do
`let msg = match channel.state() { ...`
and then after the match do 
```
let event = make_open_ack_channel_event(
                   &msg,
                   conn_id,
                   counterparty,
              );
 self.check_emitted_event(event)
       .map_err(|e| Error::IbcEvent(e.to_string()))
 ```

Reply:
Thank you! I've tried cleaning the match block to check the message then the state changes.